Mining improvements and changes- Pt .2: Generation and Mobs

Sun-Soaked opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->
Yes I know I finished part 2 before part 1 was even fully merged, it just be like that sometimes.

## About The Pull Request
- Adds gemstones! You can buy stuff by selling them, or hoard them like a dragon. More functionality pending, maybe.
- There's a [BIG SHOT] in all of us, if you can find the mask of riches...
- The necropolis is stirring, and the tendrils it punches through earth have grown far more formidable. 
- A strange demonic incursion can now be found across the surface of ice worlds- the rewards for destroying these structures can be quite considerable, but beware- that which exits may not be happy to see you. The new portals have a slight tendency to pull objects across time and space.
- Adds and nerfs both ice whelps and ice demons.
- Asteroid spawners now drop ore, as well as occasional scraps of mining paraphernalia.
- Adds crusher trophies to the polar bear and white wolf. 
- Puce now grows everywhere, and fireblossoms now grow on lavaland.
- Boosts mob elite spawn chances slightly, and considerably boosts drop chance for elite trophies. These used to be nearly unobtainable, but can now generally be acquired in-round.
- Crusher users will rejoice, for ancient goliaths and dwarf legion now have crusher trophies. The trophy for the normal goliath now works as intended. 
- Adds three new elites: The Polar Warbear, Alpha Wolf, and Primordial demon. ...enjoy.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Generally amplified the rewards and compounded the danger of both spawner types, alongside my other mining PR- this supports our current setup of unlimited possible rewards, and is setting up for a world where smashing tendrils and mugging megafauna for their lunch money is a team effort.
Asteroid good
Other assorted mining gen improvements.

## Changelog
:cl:
add: Adds a wolf talisman that can be crafted from both wolf trophies.
add: The legion staff, allowing one to summon legion skulls to their side.
tweak: makes legion and wolf trophies drop independent of crusher usage.
balance: Nerfs money gained from selling necropolis loot.
add: gemstones! They can be sold on export for money, or held onto I guess. Adds rare chances for them to drop to a few mobs, and to a particular demonic pool.
add: A particularly BIG mask that alters speech like the Italian moustache.
balance: Reduces the health and functional speed/range of whelps and ice demons.
balance: increases the health and maximum spawns of lava tendrils.
tweak: Makes asteroid spawners drop ore as well, with a small chance of other loot.
add: implements demonic portals on snow planets. 
add: Ports the white wolf and polar bear crusher trophies.
tweak: significantly enriched demonic portal loot drops and added scaling difficulty- You are not safe when it dies, so be ready to fight off whatever comes through.
tweak: adds on-contact hostile versions of clown mobs.
tweak: non-clowns can now wear the Cosmohonk hardsuit.
tweak: adds puce to all 3 high-difficulty planets. Adds fireblossom to the lavaland flora pool.
tweak: Increases elite spawnchance slightly, and significantly increases chances for elites to drop their unique crusher trophy
add: A new polar bear elite, a wolf elite, a demon elite and a trophy alongside each. 
tweak: Adds unique trophies for the ancient goliath, dwarf legion, and ash drake(old ash drake trophy is now the ice whelp trophy)
fix: goliath tentacle crusher trophy now works as intended, instead of doing nothing.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sun-Soaked committed:
monke

jungool

Sun-Soaked committed:
Merge pull request #2 from MemedHams/memedhams-monke

Pan troglodytes, also known as the common chimpanzee

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into master

Sun-Soaked committed:
Revert "Pan troglodytes, also known as the common chimpanzee"

Sun-Soaked committed:
Merge pull request #5 from MemedHams/revert-2-memedhams-monke

Revert "Pan troglodytes, also known as the common chimpanzee"

Sun-Soaked committed:
Iceworld 2: revenge of more ice

-Ice planets now have netheric portals, with a new loot table
-The illusive and dangerous ice demons have moved into iceworlds, alongside ash whelps.
-Slightly nerfed both of the above.
-Considerably boosted the relative danger of legion tendrils, nearly doubling their health and increasing their max mob deploys. If you want that sweet, sweet [hyperlink blocked] you're gonna have to work for it

Sun-Soaked commented:
ignore the monke stuff, I fucked up my base fork and had to fix it

ZephyrTFA commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -133,17 +137,21 @@ GLOBAL_LIST_INIT(ore_probability, list(
+			new /mob/living/simple_animal/hostile/netherworld/blankbody(loc)
 		if(6)
 			new /obj/item/rod_of_asclepius(loc)
+			new /obj.effect/mob_spawn/human/doctor(loc)
```
```suggestion
			new /obj/effect/mob_spawn/human/doctor(loc)
```

ZephyrTFA commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -152,41 +160,93 @@ GLOBAL_LIST_INIT(ore_probability, list(
 			new /mob/living/simple_animal/hostile/megafauna/blood_drunk_miner/doom(loc)
+		if(29)
+			new /mob/living/simple_animal/hostile/megafauna/swarmer_swarm_beacon(loc)//god help you
+		if(30)
```
All of these right here are a hard no

Sun-Soaked replied:
The swarmer beacon now drops a small stack of swarmers and an infiltrator engi, and the syndicate now has if/either logic for either reward

Sun-Soaked committed:
passby fun

-Turns new quixote into a sub-item
-Reworks a particularly harsh portal drop

ZephyrTFA commented on code/game/objects/structures/lavaland/necropolis_tendril.dm:
```diff
@@ -7,8 +7,8 @@
 	faction = list("mining")
-	max_mobs = 3
-	max_integrity = 250
+	max_mobs = 5//shiptest edit, boosted power and health
```
Comment isn't needed

Sun-Soaked committed:
Fuck

yeah uh hmm that's

Co-authored-by: Zephyr <12817816+ZephyrTFA@users.noreply.github.com>

Sun-Soaked committed:
adds prob chance to syndicate netherworld drop

I do as you command

Sun-Soaked committed:
removes highly salacious comment

scandalous

Sun-Soaked committed:
Merge branch 'planetgen' of https://github.com/MemedHams/Shiptest into planetgen

Sun-Soaked committed:
additional loot stuff

adds more loot to pool, adds significantly more even distribution of mobs.

Sun-Soaked committed:
whoopsies

yeah I forgot this

Sun-Soaked committed:
more looter stuff

-reduces the probability of just getting 5 stacks of iron when killing an ice cave, increasing the relative value

-adds an upgrade scroll to the wizard summon. Practically useless but funny

Sun-Soaked committed:
I have no idea how this happened

vsc autofill you son of a seabiscuit

Sun-Soaked committed:
cest un braindead

ooh la la

Sun-Soaked committed:
Ice is nice

-fixes the mess with spawn chances(I think)
-Adds a bunch of QOL and new stuff to the demon drops
-slightly un-nerfs ice whelps

Sun-Soaked committed:
passby 2, passby through

-long-overdue polar bear/wolf nerfs
-adds yog's necroseed, polishes it up a little, adds to demonic portal pool
-BDM dropped by demonic portal is no longer literally the worst
-butcher gloves

Sun-Soaked committed:
ice whelp aaaa I AM ON FIRE AAAAAA

INCREASES ICE WHELP FIRE COOLDOWN TIME AND RESTORES RANGE NOW PUT ME OUT I AM ACTUALLY ON FIRE HOLY SHIT

Sun-Soaked committed:
trinkets and baubles, see how I glitter

-crusher trophies for the polar bear and white wolf
-asteroid burrows now drop ore, with a chance to drop assorted mining loot
-reworks asteroid generation slightly to be more stuff-ified (I think)

Sun-Soaked committed:
passby too many, this one is good though

-significant rework to demonic portal spawn logic, adding both percentile chances of loot and variable difficulty based on amount of loot spawned.
-removes cosmohonk limitations, since the original clown guild died out in the honk wars
-adds immediately hostile versions of clown mobs
-adds other loot to cave drops
-reskins and buffs giant legion slightly
-adds research_note loot versions for mapping
-adds a looted version of the rewind camera, with 5 uses
-fixes the DME

Sun-Soaked committed:
I broke it

-AAAAAAAA

Sun-Soaked committed:
bugfix 2

what writing spawn lists for 5 hours straight does to a MF

Sun-Soaked committed:
balance tweaks

- I was a little too harsh with some chances
-adds some new possible drops

Sun-Soaked committed:
adds more loot

-Spamton mask

-Assorted hivelord stuff

Sun-Soaked committed:
[hyperlink blocked]

-adds withdraw savings ability for spamton. Warning: Does not include shipping or handling.
-fixes some scunthorpe problem examples

Sun-Soaked committed:
fixes again

-So much [hyperlink blocked] to get rid of

Sun-Soaked committed:
when the [[kronor]] is [4.99]]

allows internals for spamton mask

Sun-Soaked committed:
trinkets and baub2

-Adds gemstones! They are pretty, and can be sold for a good price
-Adds gemstones rarely to a few monster and loot drops.
-nerfs money gain from lavaland treasures. Unfortunate but inevitable

Sun-Soaked commented:
WAAAAA

Sun-Soaked committed:
when the upstream merges are incompatible!

Indescribable suffering

Sun-Soaked committed:
a

b

Sun-Soaked committed:
Merge pull request #6 from shiptest-ss13/master

merge upstream guh

Sun-Soaked committed:
this will fix it

I am unamicable towards git of hub

Sun-Soaked committed:
dwake

-Adds draconic amber to ash dragon loot

Sun-Soaked committed:
supremely lootable

-I reliably got nothing/only mobs from these pools, so I added some additional stuff to them to pad the probability

Sun-Soaked committed:
clowning around

-adds a fun new clown mob
-implements fun new clown mob

Sun-Soaked committed:
BY THE POWER OF LEGIONSKULL

-increases elite spawn chance slightly
-increases chance for elites to drop their crusher trophy(it was, like, silly low before consider how rare they are)
-adds new crusher trophies for the Ash Drake, Dwarf Legion, Ancient Goliath
-adds an elite variant of the polar bear, with a new trophy.
-fixes the goliath trophy not working as intended
-reflavours the Kinetic crusher to induce magnetism
-fixes the mess with different sub-trophies being incompatible

Sun-Soaked commented:
this WILL become a merge conflict when the mining rework PR is merged, I'm almost sure of it

Anderson-Cooper commented on code/modules/mob/living/simple_animal/hostile/clown.dm:
```diff
@@ -310,3 +310,27 @@
 	armour_penetration = 30
 	environment_smash = ENVIRONMENT_SMASH_WALLS//no escape
+
+/mob/living/simple_animal/hostile/clown/stacked
```
i opened this at school and now i am frightened. I want my mom to pick me up early.

Sun-Soaked replied:
join in the fun!join in the fun!join in the fun!

Anderson-Cooper replied:
stop, i am filled with fear

Sun-Soaked committed:
pass, bye

-Adds an elite ice demon(original ice demon robustness with added health and damage, have fun)
-Fixes some missing context stuff

Sun-Soaked committed:
passby of the passby

-the elite ice demon now drops a unique trophy
-fixes something I forgor

Sun-Soaked committed:
-bruhfs

-buffs some elites

Sun-Soaked committed:
widdle iddle baby fix

waaaa

Sun-Soaked committed:
fixes and mixes

-fixes polar bear rage, nerfs health slightly to compensate
-ports charging mechanics, they can now be implemented for any mobs.
-adds a charge to the wolf and considerably reduces their speed.

Sun-Soaked committed:
wolf fixes

a

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into planetgen

Sun-Soaked committed:
nyappear for your court summons.

-Adds the Alpha Wolf, a mini-boss level elite that spawns extremely rarely.
-Adds an Alpha Wolf crusher trophy.
-Adds the hunters talisman, an item crafted from the rare trophy drops created by both wolf types.
-adds the Legion Staff, which can be crafted from both types of skull, a regen core, and some bone. Summon the legions to defeat your enemies!
-changes the skull codpiece recipe to require a legion skull, but buffs it slightly
-adds verbage for some rare drops

Sun-Soaked committed:
Merge branch 'planetgen' of https://github.com/MemedHams/Shiptest into planetgen

HighPrincessErinys commented:
Cursed Buisnessman's Mask is in itself a typo but I must also question why it's [KRONOR] and not [KROMER]

triplezeta commented:
probably to make it slightly less on the nose

HighPrincessErinys commented:
also, consider nerfing the ice whelps further, probably by lowering their action speed (if that makes sense) and or their health. 300 health on a thing that can give you massive firestacks and acts slightly faster than a goliath is, while also doing a flat 20 damage per hit when a goliath does 12 to 20 damage per hit randomly is. nah.

Sun-Soaked commented:
they *are* considerably slower than they used to be, already- but it's important to note that their skin gives a guaranteed path to making one of the strongest armors in the game(this is just how they already were) so I don't want to push them too low. They're not quite as ruthless now, and their fire attacks happen less frequently, and they spawn less often- but further than that, I'm not sure about pushing them lower

Sun-Soaked committed:
this one was not intentional whoops

(the one in the actual voice lines is)

HighPrincessErinys commented:
at the very (extra) least give their attacks some rng, even if its more high end than goliaths. say, 16-20 just for a little variety in your mauling experience

Sun-Soaked committed:
-minor buff to legion staff

what I really need is to make it launch swarms, but till then this will at least make it viable

Sun-Soaked committed:
-rewritten legion skull logic

thanks @azarak

Sun-Soaked committed:
removes unfinished earthquake

-ok this is a little silly, gotta integrate this into my next project instead

Sun-Soaked committed:
aaaaa

It's been a long day

Sun-Soaked committed:
fixes a watcher rare gem drop, flavors it

see above

Sun-Soaked committed:
slightly buffs magmawing watchers

-they did basically poke damage before for some reason

Sun-Soaked committed:
legion pillar and sell prices

-The skulls can now be used to create a flex item of incredible prowess, the legion on a stick. Fill your ship with these to prove that you're the real one.
-adds sell prices for all the trophies, impending them having real uses.

Sun-Soaked committed:
White, sands

-adds two new spellbooks
-reworks whitesands human mobs slighty and changes their flavor, gives them a crusher trophy, and implements them on whitesand
-shaves down KA text
-adds two new legion corpses
-fixes the outfit on survivors, and buffs it

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into planetgen

Sun-Soaked committed:
advanced rng

-syndicate moment

Sun-Soaked committed:
guh

whoopsies

Sun-Soaked committed:
violencia

-I am booboo the fool

Sun-Soaked committed:
bugfixes too many help

aaaaaAAAAAA

Sun-Soaked committed:
bugfixes 2

suffering

Sun-Soaked committed:
dwagonne my beloved

-Adds a crafting use for the Ash Spike, makes it always drop from the drake.
-Adds the Dragonslayer's Spear, said crafting use. This extremely robust spear also sets those hit with it on fire.

Sun-Soaked committed:
more [[hyperlink blocked]] word replacements

that wasn't very [big shot] of you

TheNeoGamer42 commented on code/modules/cargo/exports/lavaland.dm:
```diff
@@ -72,3 +107,34 @@
 	export_types = list(/obj/item/clothing/head/helmet/space/hostile_environment)
+
+//not technically lavaland but this had a useful infrastructure to store them under
+/datum/export/lavaland/gems/diamond
```
uuuuuuuuuuhhhhhhhhhhhhhhhh this is the same thing as the thing below it which will probably break shit?

Sun-Soaked replied:
fixed

Sun-Soaked committed:
survivors now fight mobs

-guh

Sun-Soaked committed:
fix

Sun-Soaked committed:
increases rarity of whitesand hermits

-when the squikky go bop
